src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs

```diff
@@ -2330,5 +2330,219 @@ public void M()
+                // (14,13): error CS8793: The given expression always matches the provided pattern.
+                //         _ = 1 is >= 0; // 10
+                Diagnostic(ErrorCode.WRN_GivenExpressionAlwaysMatchesPattern, "1 is >= 0").WithLocation(14, 13),
+                // (15,13): error CS8793: The given expression always matches the provided pattern.
+                //         _ = 1 is >= 1; // 11
+                Diagnostic(ErrorCode.WRN_GivenExpressionAlwaysMatchesPattern, "1 is >= 1").WithLocation(15, 13),
+                // (16,13): warning CS8519: The given expression never matches the provided pattern.
+                //         _ = 1 is >= 2; // 12
+                Diagnostic(ErrorCode.WRN_GivenExpressionNeverMatchesPattern, "1 is >= 2").WithLocation(16, 13)
+                );
+        }
+
+        [Fact]
+        public void RelationalPatternInSwitchWithConstantControllingExpression()
+        {
+            var source =
+@"public class C
+{
+    public void M()
+    {
+        switch (1)
+        {
+            case < 0: break; // 1
+            case < 1: break; // 2
+            case < 2: break;
+            case < 3: break; // 3
+        }
+    }
+}
+";
+            CreateCompilation(source, options: TestOptions.ReleaseDll, parseOptions: TestOptions.RegularWithPatternCombinators).VerifyDiagnostics(
+                // (7,23): warning CS0162: Unreachable code detected
+                //             case < 0: break; // 1
+                Diagnostic(ErrorCode.WRN_UnreachableCode, "break").WithLocation(7, 23),
+                // (8,23): warning CS0162: Unreachable code detected
+                //             case < 1: break; // 2
+                Diagnostic(ErrorCode.WRN_UnreachableCode, "break").WithLocation(8, 23),
+                // (10,23): warning CS0162: Unreachable code detected
+                //             case < 3: break; // 3
+                Diagnostic(ErrorCode.WRN_UnreachableCode, "break").WithLocation(10, 23)
+                );
+        }
+
+        [Fact]
+        public void RelationalPatternInSwitchWithOutOfRangeComparand()
+        {
+            var source =
+@"public class C
+{
+    public void M(int i)
+    {
+        switch (i)
+        {
+            case < int.MinValue: break; // 1
+            case <= int.MinValue: break;
+            case > int.MaxValue: break; // 2
+            case >= int.MaxValue: break;
+        }
+    }
+    public void M(uint i)
+    {
+        switch (i)
+        {
+            case < 0: break; // 3
+            case <= 0: break;
+            case > uint.MaxValue: break; // 4
+            case >= uint.MaxValue: break;
+        }
+    }
+}
+";
+            CreateCompilation(source, options: TestOptions.ReleaseDll, parseOptions: TestOptions.RegularWithPatternCombinators).VerifyDiagnostics(
+                    // (7,18): error CS8120: The switch case has already been handled by a previous case.
```

Comment:
>error CS8120: The switch case has already been handled by a previous case. [](start = 31, length = 74)

This diagnostic message is confusing since there is no previous case.

Reply:
Not sure what to do about this, as this is the message used (since 7) when a switch case is unreachable. What do you suggest?

---
In reply to: [395883716](https://github.com/dotnet/roslyn/pull/42501#discussion_r395883716) [](ancestors = 395883716)

Reply:
Or perhaps "Unreachable case. If there are values that satisfy this case, those values have been handled in previous cases."

---
In reply to: [395908208](https://github.com/dotnet/roslyn/pull/42501#discussion_r395908208) [](ancestors = 395908208,395883716)